### ISSUE TITLE
Remove header rewriter unnecessary options

### ISSUE THREAD

**Issue by ayrtonm** [immunant/IA2-Phase2#158]
tl;dr: #104 added flags to the rewriter to handle cases where headers are shared between compartments. #149 made `--omit-wrapper` unnecessary so we could remove it. `--shared-headers` also became unnecessary after the changes in #114.

[This list](https://github.com/immunant/IA2-Phase2/issues/104#issuecomment-1084876277) is outdated so here's an updated list of what the rewriter should do.

1. Generate a linker args file from function declarations
2. Generate call gate wrappers for direct calls from function declarations and add them to the shim source
3. Rewrite function pointer types as opaque structs
4. Generate call gate wrapper macros from function pointer types and add them to the output header
5. `#include` ia2.h and the output header (if any) in headers with our opaque structs

The rewriter does all five by default (as of #157). In some cases, functions in a header don't all belong to one compartment so `--function-allowlist=file` was added to allow skipping fn declarations selectively. In other cases, headers are only needed for the opaque pointer types so `--omit-wrappers` allowed skipping all fn declarations.

As of #149 `--omit-wrappers` isn't strictly necessary since function declarations only generate the files in points 1 and 2 and never trigger any changes in the headers (i.e. we don't add `.symver` to the headers anymore). It would be simpler to only keep the `--function-allowlist=file` option for when a subset of the function declarations should appear in the shim. For cases where all compartment transitions are through function pointers we can remove the `--omit-wrappers` option and either

1. just ignore the shim source and linker args that we generate or
2. make the `wrapperOutputFilename` argument optional and use that to determine whether to skip all function declarations or not

As for `--shared-headers`, I can't recall why we didn't get rid of it in #114, but it was unnecessary even then so we should remove it.
